src/Microsoft.ApplicationInsights.WorkerService/ApplicationInsightsExtensions.cs

```diff
@@ -89,6 +90,7 @@ public static IServiceCollection AddApplicationInsightsTelemetryWorkerService(th
             {
                 if (!IsApplicationInsightsAdded(services))
                 {
+                    services.AddSingleton<ITelemetryInitializer, AzureWebAppRoleEnvironmentTelemetryInitializer>();
```

Comment:
You're adding something that depends on HTTP (IHttpContextAcessor) to the WorkerService nuget...?
That seems misplaced. Can you explain?

Reply:
`AzureAppServiceRoleNameFromHostNameHeaderInitializer` is the name of the new initializer which relies on httpcontext.`AzureWebAppRoleEnvironmentTelemetryInitializer` is the old implementation relying purely on ENV variables, and it is continued to be used in WorkerService.